mycroft/skills/skill_manager.py

```diff
@@ -303,23 +361,37 @@ def run(self):
 
         self.status.set_alive()
 
-        if self.skills_config.get("wait_for_internet", True):
-            while not connected() and not self._connected_event.is_set():
-                sleep(1)
-            self._connected_event.set()
-
         self._load_on_startup()
 
-        # Sync backend and skills.
-        # why does selene need to know about skills without settings?
-        if is_paired():
-            self.manifest_uploader.post_manifest()
+        if self.skills_config.get("wait_for_internet", False):
+            LOG.warning("`wait_for_internet` is a deprecated option, update to "
+                        "specify `network_skills` and `internet_skills` in "
+                        "`ready_settings`")
+            # NOTE - self._connected_event will never be set
+            # if PHAL plugin is not running to emit the connected events
+            while not self._connected_event.is_set():
+                sleep(1)
+            LOG.debug("Internet Connected")
+        if "network_skills" in self.config.get("ready_settings"):
+            if self._network_loaded.wait(300):
+                LOG.debug("Network skills loaded")
+            else:
+                LOG.error("Gave up waiting for network skills to load")
```

Comment:
should this emit a new bus message, it could be consumed by shell/homescreen to display some status widget